src/Components/Server/src/Circuits/CircuitFactory.cs

```diff
@@ -8,6 +8,10 @@ namespace Microsoft.AspNetCore.Components.Server.Circuits
 {
     internal abstract class CircuitFactory
     {
-        public abstract CircuitHost CreateCircuitHost(HttpContext httpContext, IClientProxy client);
+        public abstract CircuitHost CreateCircuitHost(
+            HttpContext httpContext,
+            IClientProxy client,
+            string uriAbsolute,
+            string baseUriAbsolute);
```

Comment:
How do these things work? Can these be derived from the `httpContext`?

Reply:
In the prerrendered case yes, in the circuit connection they get passed in by StartCircuit on ComponentHub

Reply:
As in the circuit case BrowserUrl != RequestUrl

Reply:
OK. I'm not to drill into details like this at this point - it's internal and I think we'll end up iterating more.